frontend/src/concepts/pipelines/apiHooks/mlmd/__tests__/useGetLinkedArtifactsByEvents.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { testHook, standardUseFetchState } from '~/__tests__/unit/testUtils/hooks';
+import { MetadataStoreServicePromiseClient, Event, Artifact } from '~/third_party/mlmd';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { useGetLinkedArtifactsByEvents } from '~/concepts/pipelines/apiHooks/mlmd/useGetLinkedArtifactsByEvents';
+import { GetArtifactsByIDResponse } from '~/third_party/mlmd/generated/ml_metadata/proto/metadata_store_service_pb';
+
+// Mock the usePipelinesAPI hook
+jest.mock('~/concepts/pipelines/context', () => ({
+  usePipelinesAPI: jest.fn(),
+}));
+
+// Mock the MetadataStoreServicePromiseClient
+jest.mock('~/third_party/mlmd', () => {
+  const originalModule = jest.requireActual('~/third_party/mlmd');
+  return {
+    ...originalModule,
+    MetadataStoreServicePromiseClient: jest.fn().mockImplementation(() => ({
+      getArtifactsByID: jest.fn(),
+    })),
+    GetArtifactsByIDRequest: originalModule.GetArtifactsByIDRequest,
+  };
+});
+
+describe('useGetLinkedArtifactsByEvents', () => {
+  const mockClient = new MetadataStoreServicePromiseClient('');
+  const mockUsePipelinesAPI = jest.mocked(
+    usePipelinesAPI as () => Partial<ReturnType<typeof usePipelinesAPI>>,
+  );
+  const mockGetArtifactsByID = jest.mocked(mockClient.getArtifactsByID);
+
+  const mockArtifact = new Artifact();
+  mockArtifact.setId(1);
+  mockArtifact.setName('artifact1');
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockUsePipelinesAPI.mockReturnValue({
+      metadataStoreServiceClient: mockClient,
+    });
+  });
+
+  it('should return an empty array if no events have artifact IDs', async () => {
+    const renderResult = testHook(useGetLinkedArtifactsByEvents)([]);
+
+    expect(renderResult.result.current).toStrictEqual(standardUseFetchState([]));
+    expect(renderResult).hookToHaveUpdateCount(1);
+  });
```

Comment:
Need to wait for an update before you can assert the final value when using `useFetchState`